### PR TITLE
AArch64: Implement atomic add and swap codegen support

### PR DESCRIPTION
**Pull request by Akira1Saitoh**
Add support for inlining the atomic intrinsics on aarch64.
- atomicAddSymbol
- atomicFetchAndAddSymbol
- atomicSwapSymbol

Issue: https://github.com/eclipse/omr/issues/2958

Signed-off-by: Akira Saitoh <saiaki@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
Jenkins build aarch64

**Comment by knn-k** [comment]
Jenkins build aarch64

**Comment by knn-k** [comment]
Jenkins build aarch64

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -1033,3 +1039,371 @@ OMR::ARM64::TreeEvaluator::generateVFTMaskInstruction(TR::CodeGenerator *cg, TR:
...
+   /*
+    * Generating non-intuitive instruction sequence which uses load exclusive register 
+    * and store release exclusive register followed by full memory barrier.
+    *
+    * Because this atomic add has `volatile` semantics,
+    * no loads/stores before this sequence can reordred after it and
+    * no loads/stores after it can reordered before it.
+    *
+    * loop:
+    *    ldxrx   oldValue, [addressReg]
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp:1085]: `oldValue` here should be `oldValueReg`.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -1033,3 +1039,371 @@ OMR::ARM64::TreeEvaluator::generateVFTMaskInstruction(TR::CodeGenerator *cg, TR:
...
+    *    addx    newValueReg, oldValueReg, valueReg
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp:1220]: The same comment as above.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 3]
```diff
@@ -1033,3 +1039,371 @@ OMR::ARM64::TreeEvaluator::generateVFTMaskInstruction(TR::CodeGenerator *cg, TR:
...
+   /*
+    * Generating non-intuitive instruction sequence which uses load exclusive register 
+    * and store release exclusive register followed by full memory barrier.
+    *
+    * Because this atomic add has `volatile` semantics,
+    * no loads/stores before this sequence can reordred after it and
+    * no loads/stores after it can reordered before it.
+    *
+    * loop:
+    *    ldxrx   oldValue, [addressReg]
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp:1219]: The same comment as above.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 4]
```diff
@@ -1033,3 +1039,371 @@ OMR::ARM64::TreeEvaluator::generateVFTMaskInstruction(TR::CodeGenerator *cg, TR:
...
+   /*
+    * Generating non-intuitive instruction sequence which uses load exclusive register 
+    * and store release exclusive register followed by full memory barrier.
+    *
+    * Because this atomic swap has `volatile` semantics,
+    * no loads/stores before this sequence can reordred after it and
+    * no loads/stores after it can reordered before it.
+    *
+    * loop:
+    *    ldxrx   oldValue, [addressReg]
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp:1336]: The same comment as above.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp (hunk 1 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: fixed.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp (hunk 3 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: fixed it too.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp (hunk 4 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: fixed.

**Comment by knn-k** [comment]
Jenkins build aarch64

**Review by knn-k — APPROVED** [review]

**Pull request merged**